lib/aad_oauth.dart

```diff
@@ -17,8 +17,7 @@ class AadOAuth {
   RequestToken _requestToken;
 
   factory AadOAuth(config) {
```

Comment:
and since we want to store multiple tokens in secure storage, we need to set different identifiers/keys:
``` dart
AadOAuth(Config config) {
    _config = config;
    _authStorage = new AuthStorage(tokenIdentifier: config.tokenIdentifier);
    _requestCode = new RequestCode(_config);
    _requestToken = new RequestToken(_config);
  }
```

and for lib/helper/auth_storage.dart

``` dart
class AuthStorage {
  static AuthStorage shared = new AuthStorage();
  FlutterSecureStorage _secureStorage = FlutterSecureStorage();
  String _tokenIdentifier;

  AuthStorage({String tokenIdentifier = "Token"}) {
    _tokenIdentifier = tokenIdentifier;
  }

  Future<void> saveTokenToCache(Token token) async {
    var data = Token.toJsonMap(token);
    var json = Convert.jsonEncode(data);
    await _secureStorage.write(key: _tokenIdentifier, value: json);
  }

  Future<T> loadTokenToCache<T extends Token>() async {
    var json = await _secureStorage.read(key: _tokenIdentifier);
    if (json == null) return null;
    try {
      var data = Convert.jsonDecode(json);
      return _getTokenFromMap<T>(data);
    } catch (exception) {
      print(exception);
      return null;
    }
  }

  Token _getTokenFromMap<T extends Token>(Map<String, dynamic> data) =>
      Token.fromJson(data);

  Future clear() async {
    _secureStorage.delete(key: _tokenIdentifier);
  }
}
```

lib/model/config.dart

``` dart
class Config {
  final String azureTenantId;
  final String azureTenantName;
  final String userFlow;
  String authorizationUrl;
  String tokenUrl;
  final String clientId;
  final String clientSecret;
  final String redirectUri;
  final String responseType;
  final String contentType;
  final String scope;
  final String resource;
  final bool isB2C;
  Rect screenSize;
  String userAgent;
  String tokenIdentifier;

  Config(
    this.azureTenantId,
    this.clientId,
    this.scope,
    this.redirectUri, {
    this.isB2C = false,
    this.azureTenantName,
    this.userFlow,
    this.clientSecret,
    this.resource,
    this.responseType = "code",
    this.contentType = "application/x-www-form-urlencoded",
    this.screenSize,
    this.userAgent,
    this.tokenIdentifier = "Token",
  }) {
    this.authorizationUrl = isB2C
        ? "https://$azureTenantName.b2clogin.com/$azureTenantName.onmicrosoft.com/oauth2/v2.0/authorize"
        : "https://login.microsoftonline.com/$azureTenantId/oauth2/v2.0/authorize";
    this.tokenUrl = isB2C
        ? "https://$azureTenantName.b2clogin.com/$azureTenantName.onmicrosoft.com/$userFlow/oauth2/v2.0/token"
        : "https://login.microsoftonline.com/$azureTenantId/oauth2/v2.0/token";
  }
}
```

and in the b2c example main file add different tokenIdentifiers to the configs